PWGHF/HFC/TableProducer/correlatorDsHadrons.cxx

```diff
@@ -602,6 +602,18 @@ struct HfCorrelatorDsHadrons {
           if (std::abs(particleAssoc.eta()) > etaTrackMax || particleAssoc.pt() < ptTrackMin || particleAssoc.pt() > ptTrackMax) {
             continue;
           }
+
+          if (particle.daughtersIds().size() != 3) {
```

Comment:
`daughterIds()` has always size 2 and it contains the ids of the first and the last daughters only. You should in principle loop from the first and the last, but also in this case it would not work in your case because the daughters would be a pion and a phi meson, so you should loop iteratively up to the final state particles. I would suggest you to use the `getDaughters` method in `RecoDecay` to easily get them: https://github.com/AliceO2Group/O2Physics/blob/f347b6bf54e06ce7271e963ebf0d065f40781962/Common/Core/RecoDecay.h#L600

Reply:
Hi @scattaru it should be like that, provided that you set the `depth` to `2` and not `1` (see comment above). To be 100% sure though I would suggest you to test it locally with some printouts ;)

Reply:
Hi @fgrosa thanks a lot! Then I'll try to test locally with these changes and see if it works

Reply:
Hi @scattaru I think that you should also specify the pdg codes of the final daughter particles, otherwise they are never considered as `final` and they are not stored in the list of indices

Reply:
https://github.com/AliceO2Group/O2Physics/blob/30983eeaa6f821437864a0fa1d88a217d464cd76/Common/Core/RecoDecay.h#L597

Reply:
> Without specifying the PDG codes, `getDaughters` will return the list of all daughters at the given maximum depth of the decay tree.

ah thanks, I indeed overlooked that! @scattaru my last comment then should be that you should replace `std::array{0}` with an empty array `std::array{}`, since otherwise you are providing one PDG code (`0`)

Reply:
Hi @fgrosa, @vkucera, thanks a lot for your help! Now I have replaced the array with an empty one. Looking at the similar part of the code written in the candidatecreator3prong I have used the array initialized with zero. I thought that there were not problems since that if the PDG scheme is the one used in PYTHIA there should not be particles with a PDG code = 0, but I could be wrong. However to be sure it is better as you suggested Fabrizio to use the empty array.

Reply:
@scattaru `{0}` would be fine since you correctly pointed out that there is no particle with that PDG.

Reply:
@scattaru @vkucera sorry for the second wrong suggestion! You can keep it with the daughter pdgs explicitly, or with `{0}`, both are fine for me.

Reply:
Hi @fgrosa, for me it is fine the way it is implemented now